### PR TITLE
🔧 nft create form

### LINKED ISSUE #7770: create NFT form - royalty and tags not working
in create NFT form

### 1. royalty input should be visible on rmrk2 as well

image

### 2. form state not reactive to changes in royalty amount:

image

### 3. Properties also not working:

image

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #7770
- [ ] Requires deployment <snek/rubick/worker>

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)

#### Result:
https://kodadot.xyz/ksm/gallery/20222382-1AD2CE70E162F5CC33-MWFBPXKUNGGX-TEST_ROYALTY-00000051

image

image

 

### <samp>🤖 Generated by Copilot at 0d614a3</samp>

Refactored some components to use `v-model` for simpler data binding and enabled royalty support for both Basilisk and Remark NFTs.

### <samp>🤖 Generated by Copilot at 0d614a3</samp>

> _We bind the data with `v-model`_
> _We unleash the power of the `NeoField`_
> _We refactor the code to make it clean_
> _We are the masters of the `RoyaltyForm`_

### CODE REVIEW

File: components/create/CreateNft.vue [hunk 1]
```diff
@@ -144,10 +144,10 @@
       </NeoField>
 
       <!-- royalty -->
-      <NeoField v-if="isBasilisk">
+      <NeoField v-if="isBasilisk || isRemark">
```
Comment (roiLeo) [inline components/create/CreateNft.vue:147]: not available on `ksm` chain (only rmrk2)

**Review by roiLeo — APPROVED** [review]
otherwise code lgtm

File: components/create/CreateNft.vue (hunk 1 above)
Comment (daiagi) [reply components/create/CreateNft.vue]: thanks
fixed

**Comment by prury** [comment]
@daiagi I've created an NFT with custom royalties/recipients and its not showing on details here for me
i'm aware it only shows when you specify a custom royalty recipient
transaction: https://kusama.subscan.io/tx/0x83d3893e20672e6689ec538f7aea5652be474936c0a44dccec4ec38525486b2a
NFT: https://deploy-preview-7771--koda-canary.netlify.app/ksm/gallery/20237479-7EA1DCF47E98A25067-RDNWYGCTFOERX-DESERT_HUB-00000004

**Comment by prury** [comment]
only works when minting + listing https://deploy-preview-7771--koda-canary.netlify.app/ksm/gallery/20237759-7EA1DCF47E98A25067-RDNWYGCTFOERX-DESERT_VILLA_2-00000005

**Comment by daiagi** [comment]
Thanks @prury !!

found another bug in the form, entering an address to receive royalty was broken

fixed

**Review by prury — APPROVED** [review]

**Review by Jarsen136 — APPROVED** [review]

**Comment by yangwao** [comment]
Thanks!
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 4.25 USD/DOT ~ 7.059 $DOT
🧗 [1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8](https://polkadot.subscan.io/account/1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)
🔗 [0x89e684819f8b2578063ceb67882dbaa1d9f4b6750acdd198071908ebe1df5ea8](https://polkadot.subscan.io/extrinsic/0x89e684819f8b2578063ceb67882dbaa1d9f4b6750acdd198071908ebe1df5ea8)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**